### ISSUE TITLE
Tests fail under Python 3.8b1+

### ISSUE THREAD

**Issue by mgedmin** [zopefoundation/zope.traversing#13]
tox -e py38 fails:

```
py38 create: /home/mg/src/zopefoundation/zope.traversing/.tox/py38
py38 installdeps: .[test], coverage
py38 develop-inst: /home/mg/src/zopefoundation/zope.traversing
py38 installed: coverage==4.5.3,python-gettext==4.0,pytz==2019.1,six==1.12.0,transaction==2.4.0,zope.annotation==4.7.0,zope.browser==2.3,zope.browserresource==4.3,zope.component==4.5,zope.configuration==4.3.1,zope.contenttype==4.4,zope.deferredimport==4.3,zope.deprecation==4.4.0,zope.event==4.4,zope.exceptions==4.3,zope.hookable==4.2.0,zope.i18n==4.6.2,zope.i18nmessageid==4.3.1,zope.interface==4.6.0,zope.location==4.2,zope.proxy==4.3.1,zope.publisher==5.0.1,zope.schema==4.9.3,zope.security==4.3.1,zope.tales==5.0,zope.testing==4.7,zope.testrunner==5.0,-e git+git@github.com:zopefoundation/zope.traversing.git@cf8d718e589100108234fb22ae9c4cc991331c68#egg=zope.traversing
py38 run-test-pre: PYTHONHASHSEED='3467374393'
py38 run-test: commands[0] | coverage run -m zope.testrunner --test-path=src
Running zope.testrunner.layer.UnitTests tests:
  Set up zope.testrunner.layer.UnitTests in 0.000 seconds.


Failure in test traverse (zope.traversing.namespace.debug)
Failed doctest test for zope.traversing.namespace.debug.traverse
  File "/home/mg/src/zopefoundation/zope.traversing/src/zope/traversing/namespace.py", line 594, in traverse

----------------------------------------------------------------------
File "/home/mg/src/zopefoundation/zope.traversing/src/zope/traversing/namespace.py", line 656, in zope.traversing.namespace.debug.traverse
Failed example:
    adapter.traverse('source', ())
Expected:
    Traceback (most recent call last):
    ...
    ValueError: Debug flags only allowed in debug mode
Got:
    <object object at 0x7f5fb3438a80>

/home/mg/src/zopefoundation/zope.traversing/.tox/py38/lib/python3.8/site-packages/zope/publisher/http.py:399: DeprecationWarning: urllib.parse.splitport() is deprecated as of 3.8, use urllib.parse.urlparse() instead
  hostname, port = splitport(host)
  Ran 110 tests with 1 failures, 0 errors and 1 skipped in 0.348 seconds.
Tearing down left over layers:
  Tear down zope.testrunner.layer.UnitTests in 0.000 seconds.
ERROR: InvocationError for command /home/mg/src/zopefoundation/zope.traversing/.tox/py38/bin/coverage run -m zope.testrunner --test-path=src (exited with code 1)
_________________________________________ summary _________________________________________
ERROR:   py38: commands failed
```

First noticed by @icemac in https://github.com/zopefoundation/zopetoolkit/pull/36

**Comment by mgedmin** [comment]
This smells like a bug in the Python 3.8 optimizer, triggered by `if not self.enable_debug or not __debug__:`.

**Comment by mgedmin** [comment]
I've filed https://bugs.python.org/issue37269.

**Comment by icemac** [comment]
The issue seems to be fixed on the Python 3.8 branch.
TravisCI is currently green for this package, see https://travis-ci.org/zopefoundation/zope.traversing/builds/546709201

@mgedmin Are you okay with closing this issue here as resolved?

**Comment by mgedmin** [comment]
Fixed upstream.
